plugins/rbac/src/hooks/useMembers.ts

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { useAsync, useAsyncRetry } from 'react-use';
+
+import { stringifyEntityRef } from '@backstage/catalog-model';
+import { useApi } from '@backstage/core-plugin-api';
+
+import { rbacApiRef } from '../api/RBACBackendClient';
+import { MembersData } from '../types';
+import { getMembersFromGroup } from '../utils/rbac-utils';
+
+export const useMembers = (roleName: string) => {
+  const rbacApi = useApi(rbacApiRef);
+  const {
+    loading: rolesLoading,
+    value: role,
+    retry,
+  } = useAsyncRetry(async () => {
+    return await rbacApi.getRole(roleName);
+  });
+
+  const { loading: membersLoading, value: members } = useAsync(async () => {
+    return await rbacApi.getMembers();
+  });
+
+  const data = React.useMemo(
+    () =>
+      role?.[0].memberReferences.reduce((acc: MembersData[], ref) => {
```

Comment:
why do we need `0` here, you are quering based on the role name so should be just one role right? Is it  due to the response

Reply:
yes, the response is an array with one role